src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/DependenciesGraphProvider.cs

```diff
@@ -278,33 +273,26 @@ Graph IGraphProvider.Schema
                     continue;
                 }
 
-                var nodeInfo = inputGraphNode.GetValue<IDependencyNode>(
-                                    DependenciesGraphSchema.DependencyNodeProperty);
+                var nodeInfo = await GetDependencyNodeInfo(graphContext, inputGraphNode, projectPath).ConfigureAwait(false);
                 if (nodeInfo == null)
                 {
                     continue;
                 }
 
-                var subTreeProvider = await GetSubTreeProviderAsync(graphContext, 
-                                                                    inputGraphNode, 
-                                                                    projectPath,
-                                                                    nodeInfo.Id).ConfigureAwait(false);
-                if (subTreeProvider == null)
-                {
-                    continue;
-                }
+                var node = nodeInfo.Item1;
+                var subTreeProvider = nodeInfo.Item2;
 
-                if (!nodeInfo.Flags.Contains(DependencyNode.PreFilledFolderNode))
+                if (!node.Flags.Contains(DependencyNode.PreFilledFolderNode))
```

Comment:
>if (!node.Flags.Contains(DependencyNode.PreFilledFolderNode)) [](start = 16, length = 61)

Why do we conditionally refresh the node here, but unconditionally refresh it in CheckChildrenAsync? Does this check prevent us from moving the node refresh into `GetDependencyNodeInfo()`?